src/tcgwars/logic/impl/gen4/DiamondPearlPromos.groovy

```diff
@@ -613,19 +504,39 @@ public enum DiamondPearlPromos implements LogicCardInfo {
         return basic (this, hp:HP090, type:FIGHTING, retreatCost:3) {
           weakness W
           move "Nose Unit", {
-            text "This attack does 20 damage to each of your opponent�s Pok�mon that has any Pok�-Bodies."
+            text "This attack does 20 damage to each of your opponent's Pokémon that has any Poké-Bodies. (Don't apply Weakness and Resistance for Benched Pokémon.)"
             energyCost M, C
             attackRequirement {}
             onAttack {
-              damage 0
+              opp.all.each {
+                for (Ability ability : it.getAbilities().keySet()) {
+                  if (ability instanceof PokeBody) {
+                    directDamage(10, it)
+                  }
+                }
+              }
             }
           }
           move "Loud Snort", {
-            text "70 damage. �s Retreat Cost is 0 until the end of your next turn."
+            text "70 damage. Probopass Pokémon G's Retreat Cost is 0 until the end of your next turn."
             energyCost M, C, C, C
             attackRequirement {}
             onAttack {
-              damage 0
+              damage 70
+              afterDamage {
+                delayed {
+                  def eff
+                  register {
+                    eff = getter GET_RETREAT_COST, LAST, self, { h ->
+                      h.object = 0
+                    }
+                  }
+                  unregister {
+                    eff.unregister()
+                  }
+                  unregisterAfter 3
```

Comment:
Most effects should wear off after leaving the active or no longer being the same Pokémon ((d)evolve/levelup). Are you sure this is one where that isn't the case?

Reply:
You also need FALL_BACK and KNOCKOUT